torch/_export/utils.py

```diff
@@ -35,55 +39,73 @@ def _check_input_constraints_pre_hook(self, *args, **kwargs):
 
     return _check_input_constraints_for_graph(
         [node for node in self.graph.nodes if node.op == "placeholder"],
+        flat_args,
         args,
+        kwargs,
         self.range_constraints,
     )
 
 
 def _check_input_constraints_for_graph(
-    input_placeholders: List[torch.fx.Node], args, range_constraints
+    input_placeholders: List[torch.fx.Node], flat_args, args, kwargs, range_constraints
 ):
-    def check(cond, msg):
-        if not cond:
-            # TODO(avik): maybe add more context, e.g., graph signature
-            raise RuntimeError(msg)
+    def get_keystr(idx: int) -> str:
+        """For a given index into the flat_args, return a human readable string
+        describing how to access it, e.g. "*args["foo"][0].bar"
+        """
+        key_path_vals, _ = tree_flatten_with_path((args, kwargs))
```

Comment:
if all the plumbing leads us to here, why not passing key_path_vals directly? Currently the function signature looks a bit weird to pass flat_args and (args, kwargs) the same time, and later I realized that this is for error message purpose. I'd prefer something more concise.

Reply:
We could, I did it this way primarily because we don't need to tree_flatten_with_path unless specifically need to construct an error message, so it's slightly more efficient.

Upon reflection, that's a stupid reason though. If someone cares that much about performance and is using the Python version of the execution path they probably need to re-examine their life decisions

Reply:
This turned out to be trickier than I thought, for various reasons. Let's leave it as is for now; I'll follow up on it later.

Reply:
actually I figured out a small hack for the part that was tricky, updating